homeassistant/components/zwave_js/__init__.py

```diff
@@ -36,16 +38,21 @@ def register_node_in_dev_reg(
         config_entry_id=entry.entry_id,
         identifiers={(DOMAIN, client.driver.controller.home_id, node.node_id)},
         sw_version=node.firmware_version,
-        name=node.name,
-        model=node.product_type,
-        manufacturer=str(node.manufacturer_id),
+        name=node.name or node.device_config.description,
+        model=node.device_config.label or str(node.product_type),
+        manufacturer=node.device_config.manufacturer or str(node.manufacturer_id),
     )
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Set up Z-Wave JS from a config entry."""
     client = ZwaveClient(entry.data[CONF_URL], async_get_clientsession(hass))
+    hass.data[DOMAIN][entry.entry_id] = {
```

Comment:
We should only store this after we know we can connect successfully. Otherwise we raise `ConfigEntryNotReady` and don't fully clean up. That's why I had it at the end.

Reply:
well, the platform init relies on it...